elm-canvas/elm-canvas.js

```diff
@@ -12,23 +12,38 @@ customElements.define(
       this.render();
     }
 
+    static get observedAttributes() {
+      return ['canvas.width', 'canvas.height'];
+    }
+
+
+    setCanvasDimensions(w, h){
+        var devicePixelRatio = window.devicePixelRatio || 1;
+        this.canvas.style.width = w;
+        this.canvas.style.height = h;
+        this.canvas.width = w * devicePixelRatio;
+        this.canvas.height = h * devicePixelRatio;
+        this.context.scale(devicePixelRatio, devicePixelRatio);
+    }
+
     connectedCallback() {
       requestAnimationFrame(() => {
         this.canvas = this.querySelector("canvas");
         this.context = this.canvas.getContext("2d");
         this.mounted = true;
 
-        var devicePixelRatio = window.devicePixelRatio || 1;
-        this.canvas.style.width = this.canvas.width;
-        this.canvas.style.height = this.canvas.height;
-        this.canvas.width = this.canvas.width * devicePixelRatio;
-        this.canvas.height = this.canvas.height * devicePixelRatio;
-        this.context.scale(devicePixelRatio, devicePixelRatio);
+        setCanvasDimensions(this.canvas.width, this.canvas.height);
```

Comment:
```suggestion
        this.setCanvasDimensions(this.width, this.height);
```
Something like this 👆. I'm not sure if those come as numbers, so we may need to do `parseInt(this.width, 10)` if they are strings. And if `this.width` doesn't work, use `this.getAttribute('width')` instead.

Also, `this.` was missing.

Reply:
This is because we want to set the width and height on the elm-canvas instead, so that attribute changed and observed attributes works well. See PR comment